### PR TITLE
NixOS snapper integration

### PR DESCRIPTION
**Pull request by ck3d**
Integrated snapper into NixOS

###### Motivation for this change
Integrate the existing snapper package into NixOS

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by 0xABAB** [comment]
@ck3d Looks great, except for the TODO. That TODO should be a separate issue cross-linked with the source code.

**Comment by ck3d** [comment]
I will transform the TODO into a description for the user, because he or she needs to create the .snapshot subvolume. Next problem I detected is following. Snapper also allows a path to a subvolume which is not a mount point. I will change my proposal to point that out.
One thing more I need to do, Snapper allows filters, which are applied to diff listings.

**Comment by ck3d** [comment]
I am done and waiting for comments.

File: nixos/modules/services/misc/snapper.nix [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        Subvolume configuration
+      '';
+
+      type = types.attrsOf (types.submodule {
+        options = {
+          subvolume = mkOption {
+            type = types.path;
+            description = ''
+              Path of the subvolume or mount point.
+              This path is a subvolume and has to contain a subvolumed named
```
Comment (0xABAB) [inline nixos/modules/services/misc/snapper.nix]: subvolumed -> subvolume?

**Review by 0xABAB — APPROVED** [review]
I just saw this one spelling error, but otherwise this should be merged.

**Comment by ck3d** [comment]
Thanks, I fixed it.

**Review by 0xABAB — APPROVED** [review]

**Comment by ck3d** [comment]
The last build fail is unrelated to my last typo fix. I do not know how to re-trigger a build.
What can I do, to get this into the master?

**Comment by Mic92** [comment]
Travis is not a blocker here. Somebody has to review it and merge it. I have not btrfs to test this though.

**Comment by ck3d** [comment]
0xABAB approved it. I think there is no need for human testing, because I wrote a NixOS test which passes.

**Comment by Mic92** [comment]
I don't think that 0xABAB has actually tested the package. I could not find core maintainers using `btrfs`, so I just relied on the tests.

**Comment by ck3d** [comment]
Thanks a lot! I tested it on three different systems.
Best regards
Christian

Am 16. Juli 2017 11:09:25 MESZ schrieb "Jörg Thalheim" <notifications@github.com>:
>I don't think that 0xABAB has actually tested the package. I could not
>find core maintainers using `btrfs`, so I just relied on the tests.

**Pull request closed without merging**